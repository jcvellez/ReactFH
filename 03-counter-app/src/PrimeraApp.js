import React, { Fragment } from "react";

const PrimeraApp = ({saludo="Hola Mundo"}) => {

    
    console.log(saludo);
    return (
        <>
            <h1>{saludo}</h1>
            <p>Mi primera aplicacion</p>
                 
        </>
    );

}

export default PrimeraApp;