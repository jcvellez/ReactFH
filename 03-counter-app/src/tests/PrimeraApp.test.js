import React from 'react'
import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import { shallow } from 'enzyme';

describe('Prueba en <PrimeraApp />', () => {
    // test('Debe mostar "Hola, soy Juank"', () => { 
    //     const saludo='Hola, soy Juank';        
    //     const {getByText}= render(<PrimeraApp saludo={saludo}/>);
    //     expect(getByText(saludo)).toBeInTheDocument();        
    //  })
    test('Debe mostrar <PrimeraApp/> correctamente', () => {
        const saludo = 'Hola, soy Juank';
        const wraper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wraper).toMatchSnapshot();
    });

    test('Debe mostrar el subtitulo enviado por props', () => {
        const saludo = 'Hola, soy Juank';
        const subtitulo = "Soy un subtitulo";
        const wraper = shallow(
            <PrimeraApp
                saludo={saludo}
                subtitulo={subtitulo}
            />
        );
        const textoParrafo = wraper.find('p').text();
        //console.log(textoParrafo);
        expect(textoParrafo).toBe(subtitulo);        
    })

})