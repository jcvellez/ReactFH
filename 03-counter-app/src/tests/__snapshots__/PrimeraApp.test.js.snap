// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en <PrimeraApp /> Debe mostrar <PrimeraApp/> correctamente 1`] = `
<Fragment>
  <h1>
     
    Hola, soy Juank
     
  </h1>
  <p>
    Soy un subtitulo
  </p>
</Fragment>
`;
